### PR TITLE
BEAUTIFUL LEGEND AND PIE CHART THAT ADJUSTS TO ENTRIES!!!!!!

### PR DESCRIPTION
**Pull request by vincentbanguyen**
it doesn't account for data or anything, soley takes in entries of MoodEntry

### CODE REVIEW

File: App/Shared/EmojiWheel/EmojiPieChartSlice.swift [hunk 1]
```diff
@@ -57,7 +55,7 @@ struct EmojiPieChartSlice: View {
 
 struct EmojiPieChartSlice_Previews: PreviewProvider {
     static var previews: some View {
-        EmojiPieChartSlice(label: "Happiness", mood: Mood.happiness, center: CGPoint(x: 350, y: 200), radius: 300, startDegree: 90, endDegree: 90 + 90, isTouched: true, accentColor: .orange, separatorColor: .black, size: 600)
+        EmojiPieChartSlice(mood: Mood.happiness, center: CGPoint(x: 350, y: 200), radius: 300, startDegree: 90, endDegree: 90 + 90, isTouched: true, accentColor: .orange, separatorColor: .black, size: 600)
```
Comment (aoenth) [inline App/Shared/EmojiWheel/EmojiPieChartSlice.swift:58]: Please create new functions or break into multiple lines.

Comment (aoenth) [reply App/Shared/EmojiWheel/EmojiPieChartSlice.swift:58]:
```swift
EmojiPieChartSlice(
    mood: Mood.happiness,
    center: CGPoint(x: 350, y: 200),
    radius: 300,
    startDegree: 90,
    endDegree: 90 + 90,
    isTouched: true,
    accentColor: .orange,
    separatorColor: .black,
    size: 600
)
```

File: App/Shared/Weekly Pie Chart/WeekPieChart.swift [hunk 2]
```diff
@@ -4,28 +4,26 @@
...
+        
+        entries.enumerated().forEach {(index, data) in
+            let value = normalizedValue(index: index, data: self.entries)
+            let mood = self.entries[index].mood
             if slices.isEmpty {
-                slices.append((.init(startDegree: 0, endDegree: value * 360, mood: mood, label: label)))
+                slices.append((.init(startDegree: 0, endDegree: value * 360, mood: mood)))
             } else {
-                slices.append(.init(startDegree: slices.last!.endDegree, endDegree: (value * 360 + slices.last!.endDegree), mood: mood, label: label))
+                slices.append(.init(startDegree: slices.last!.endDegree, endDegree: (value * 360 + slices.last!.endDegree), mood: mood))
```
Comment (aoenth) [inline App/Shared/Weekly Pie Chart/WeekPieChart.swift:26]: Use optional binding to extract the `last` variable

File: App/Shared/Weekly Pie Chart/WeekPieChart.swift [hunk 3]
```diff
@@ -34,14 +32,11 @@ struct WeekPieChart: View {
     var body: some View {
         VStack {
             GeometryReader { geometry in
-                ForEach(0..<self.data.count) { i in
-                    
-                    WeekPieChartSlice(label: pieSlices[i].label, mood: pieSlices[i].mood, center: CGPoint(x: geometry.frame(in: .local).midX, y: geometry.frame(in:  .local).midY), radius: geometry.frame(in: .local).width/2, startDegree: pieSlices[i].startDegree, endDegree: pieSlices[i].endDegree, isTouched: sliceIsTouched(index: i, inPie: geometry.frame(in:  .local)), accentColor: accentColors[i], separatorColor: separatorColor, size: geometry.frame(in: .local).width)
+                ForEach(0..<self.entries.count) { i in
+                    WeekPieChartSlice(mood: pieSlices[i].mood, center: CGPoint(x: geometry.frame(in: .local).midX, y: geometry.frame(in:  .local).midY), radius: geometry.frame(in: .local).width/2, startDegree: pieSlices[i].startDegree, endDegree: pieSlices[i].endDegree, isTouched: sliceIsTouched(index: i, inPie: geometry.frame(in:  .local)), accentColor: Color(String(describing: pieSlices[i].mood)), separatorColor: separatorColor, size: geometry.frame(in: .local).width)
```
Comment (aoenth) [inline App/Shared/Weekly Pie Chart/WeekPieChart.swift:36]: Long line

File: App/Shared/Weekly Pie Chart/WeekPieChartSlice.swift [hunk 4]
```diff
@@ -31,15 +30,15 @@ struct WeekPieChartSlice: View {
...
+            .overlay(path.stroke(separatorColor, lineWidth: 3))
             .scaleEffect(isTouched ? 1.1 : 1)
             .animation(Animation.spring())
     }
 }
 
 struct PieChartSlice_Previews: PreviewProvider {
     static var previews: some View {
-        EmojiPieChartSlice(label: "Happiness", mood: Mood.happiness, center: CGPoint(x: 350, y: 200), radius: 300, startDegree: 90, endDegree: 90 + 90, isTouched: true, accentColor: .orange, separatorColor: .black, size: 600)
+        EmojiPieChartSlice(mood: Mood.happiness, center: CGPoint(x: 350, y: 200), radius: 300, startDegree: 90, endDegree: 90 + 90, isTouched: true, accentColor: .orange, separatorColor: .black, size: 600)
```
Comment (aoenth) [inline App/Shared/Weekly Pie Chart/WeekPieChartSlice.swift:41]: Long line

**Review by aoenth — CHANGES_REQUESTED** [review]

**Review by UMPUSTEN — APPROVED** [review]

**Review by vincentbanguyen — COMMENTED** [review]
niceeee

**Pull request merged**